config/initializers/wopi_startup_check.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Only check if connection works when server starts, and if WOPI is
+# enabled
+if defined? Rails::Server && ENV['WOPI_ENABLED'] == 'true'
+  missing_vars = []
+  %w(
+    WOPI_TEST_ENABLED WOPI_DISCOVERY_URL WOPI_ENDPOINT_URL USER_SUBDOMAIN
+    WOPI_SUBDOMAIN WOPI_USER_HOST
+  ).each do |var_name|
+    missing_vars << var_name if ENV[var_name].blank?
```

Comment:
`blank?` will cause you headache. Try checking for .nil? instead.

Reply:
Huh, I guess there could be a situation where value is empty string, so you have to check both.

Reply:
You could spare some lines using the mighty functional approach

` missing_vars = %w(
    WOPI_TEST_ENABLED WOPI_DISCOVERY_URL WOPI_ENDPOINT_URL USER_SUBDOMAIN
    WOPI_SUBDOMAIN WOPI_USER_HOST
  ).select{ |env_var| ENV[env_var].nil? || ENV[env_var].blank? }`

Reply:
Actually nevermind first 2 comments. Seems activesupport adds .blank? option to nil